lib/ret_web/controllers/api/v1/oauth_controller.ex

```diff
@@ -69,6 +97,30 @@ defmodule RetWeb.Api.V1.OAuthController do
     conn |> put_short_lived_cookie("ret-oauth-flow-perms-token", perms_token)
   end
 
+  defp process_twitter_oauth(conn, account, access_token, access_token_secret, twitter_user_id) do
+    # TODO deal with case where we get a user's email and may create an account
+
+    oauth_provider =
+      OAuthProvider
+      |> Repo.get_by(source: :twitter, provider_account_id: twitter_user_id)
+      |> Repo.preload(:account)
+
+    if !oauth_provider || oauth_provider.account.account_id != account.account_id do
+      (OAuthProvider |> Repo.get_by(source: :twitter, account_id: account.account_id) ||
+         %OAuthProvider{source: :twitter, account: account})
+      |> Ecto.Changeset.change(
+        provider_access_token: access_token,
+        provider_access_token_secret: access_token_secret,
+        provider_account_id: twitter_user_id
+      )
+      |> Repo.insert_or_update()
+
+      conn
+    else
+      conn |> send_resp(401, "Another account is already connected to this twitter account.")
+    end
+  end
```

Comment:
Okay, I think I'm just thoroughly confused by this logic, so you'll have to walk me through it.

Reply:
I believe I wrote `!= account.account_id` instead of `== account.account_id` -- oops. the inner `if` block should run if there's no oauth provider for this account already or if there is, and it belongs to this account. (the inner if block can also be slightly refactored I think)

Reply:
Ah, okay, that clears things up 👍